### PR TITLE
Hashtable external key

### PR DESCRIPTION
**Pull request by AntonYudintsev**
In some cases calculation a hash can be expensive and application can already have it calculated.

 This doesn't change any public API functions, but allows external application to save some performance
 by using already caclulated hash in their own hashmap/hashset implementation based on internal EASTL 'hashtable' container

### CODE REVIEW

File: include/EASTL/internal/hashtable.h [hunk 1]
```diff
@@ -1263,10 +1263,15 @@ namespace eastl
 		node_type* DoAllocateNode(value_type&& value);
 		node_type* DoAllocateNode(const value_type& value);
 
-		eastl::pair<iterator, bool> DoInsertKey(true_type, const key_type& key);
-		iterator                    DoInsertKey(false_type, const key_type& key);
-		eastl::pair<iterator, bool> DoInsertKey(true_type, key_type&& key);
-		iterator                    DoInsertKey(false_type, key_type&& key);
+		eastl::pair<iterator, bool> DoInsertKey(true_type, const key_type& key, hash_code_t c);
```
Comment (rparolin) [inline include/EASTL/internal/hashtable.h]: Add a comment that describes the purpose of these overloads.

Comment (AntonYudintsev) [reply include/EASTL/internal/hashtable.h]: just to clarify - comment in code, right?

added overloads mimics current(old) API, so they kept just for compatibility reasons (if someone use it already).

Comment (rparolin) [reply include/EASTL/internal/hashtable.h]: Yes.  Comments in code as hints to users who don't know why both sets of API's exist and why they should reach for one or the other.

Comment (AntonYudintsev) [reply include/EASTL/internal/hashtable.h]: done

**Comment by rparolin** [comment]
Thanks for the PR!

**Pull request merged**